great_expectations/datasource/fluent/spark_azure_blob_storage_datasource.py

```diff
@@ -118,56 +125,44 @@ def test_connection(self, test_assets: bool = True) -> None:
 
-    def add_csv_asset(
+    def _build_data_connector(
         self,
-        name: str,
-        batching_regex: Union[re.Pattern, str],
-        container: str,
-        header: bool = False,
-        infer_schema: bool = False,
-        name_starts_with: str = "",
-        delimiter: str = "/",
-        order_by: Optional[SortersDefinition] = None,
-    ) -> CSVAsset:
-        """Adds a CSV DataAsset to the present "SparkAzureBlobStorageDatasource" object.
+        data_asset: CSVAsset,
+        abs_container: str = _MISSING,  # type: ignore[assignment] # _MISSING is used as sentinel value
+        abs_name_starts_with: str = "",
+        abs_delimiter: str = "/",
+        **kwargs,
+    ) -> None:
+        """Builds and attaches the `AzureBlobStorageDataConnector` to the asset."""
+        if kwargs:
+            raise TypeError(
+                f"_build_data_connector() got unexpected keyword arguments {list(kwargs.keys())}"
+            )
+        if abs_container is _MISSING:
+            raise TypeError(
+                f"'{data_asset.name}' is missing required argument 'abs_container'"
+            )
 
-        Args:
-            name: The name of the CSV asset
-            batching_regex: regex pattern that matches csv filenames that is used to label the batches
-            container: container name for Microsoft Azure Blob Storage
-            header: boolean (default False) indicating whether or not first line of CSV file is header line
-            infer_schema: boolean (default False) instructing Spark to attempt to infer schema of CSV file heuristically
-            name_starts_with: Microsoft Azure Blob Storage object name prefix
-            delimiter: Microsoft Azure Blob Storage object name delimiter
-            order_by: sorting directive via either list[Sorter] or "+/- key" syntax: +/- (a/de)scending; + default
-        """
-        order_by_sorters: list[Sorter] = self.parse_order_by_sorters(order_by=order_by)
-        asset = CSVAsset(
-            name=name,
-            batching_regex=batching_regex,  # type: ignore[arg-type] # pydantic will compile regex str to Pattern
-            header=header,
-            inferSchema=infer_schema,
-            order_by=order_by_sorters,
-        )
-        asset._data_connector = AzureBlobStorageDataConnector.build_data_connector(
+        data_asset._data_connector = self.data_connector_type.build_data_connector(
             datasource_name=self.name,
-            data_asset_name=name,
+            data_asset_name=data_asset.name,
             azure_client=self._get_azure_client(),
-            batching_regex=asset.batching_regex,
+            batching_regex=data_asset.batching_regex,
             account_name=self._account_name,
-            container=container,
-            name_starts_with=name_starts_with,
-            delimiter=delimiter,
-            file_path_template_map_fn=AzureUrl.AZURE_BLOB_STORAGE_WASBS_URL_TEMPLATE.format,
+            container=abs_container,
```

Comment:
@Kilo59 I am not certain of the purpose of changing the source value from `container` to `abs_container` (but sounds fine -- makes it a little more clear).  Thanks.

Reply:
The cause of these changes were the conflict between the `s3`/`gcs`, etc. `delimiter` argument and the pandas`.read_csv()` `delimiter` argument. But rather than only creating a `s3_delimiter` we added the cloud storage prefix to all of the asset level `connect_options` values.  `s3_delimiter`, `gcs_delimter`, `s3_max_keys` etc.